octobot_backtesting/independent_backtesting.py

```diff
@@ -0,0 +1,336 @@
+    # Warning: can take a long time when a lot of objects exist
+    def check_remaining_objects(self):
+        try:
+            from octobot_trading.exchanges.data.exchange_symbol_data import ExchangeSymbolData
+            from octobot_trading.exchanges.exchange_manager import ExchangeManager
+            from octobot_trading.exchanges.exchange_simulator import ExchangeSimulator
+            from octobot_trading.producers.simulator import OHLCVUpdaterSimulator
+            import gc
+            exchanges_count = len(self.exchange_manager_ids)
+            to_watch_objects = (ExchangeSymbolData, ExchangeManager, ExchangeSimulator, OHLCVUpdaterSimulator)
+            objects_references = {obj: 0 for obj in to_watch_objects}
+            expected_max_objects_references = {
+                ExchangeSymbolData: exchanges_count + 1,
+                ExchangeManager: exchanges_count + 1,
+                ExchangeSimulator: exchanges_count,
+                OHLCVUpdaterSimulator: exchanges_count
+            }
+            for obj in gc.get_objects():
+                if isinstance(obj, to_watch_objects):
+                    objects_references[type(obj)] += 1
+
+            for obj, max_ref in expected_max_objects_references.items():
+                if objects_references[obj] > max_ref:
+                    self._log_remaining_object_error(obj,
+                                                     max_ref,
+                                                     objects_references[obj])
+
+        except ImportError as e:
+            self._log_import_error()
+            raise e
+
+    def _log_remaining_object_error(self, obj, expected, actual):
+        self.logger.error(f"Too many remaining on {obj.__name__}: expected: {expected} actual {actual}")
+
+    def is_in_progress(self):
+        if self.backtestings:
+            return any([backtesting.is_in_progress() for backtesting in self.backtestings])
+        else:
+            return False
+
+    def get_progress(self):
+        if self.backtestings:
+            return min([backtesting.get_progress() for backtesting in self.backtestings])
+        else:
+            return 0
+
+    @staticmethod
+    def _get_market_delta(symbol, exchange_manager, min_timeframe):
+        from octobot_trading.api.symbol_data import get_symbol_data, get_symbol_historical_candles
+        market_data = get_symbol_historical_candles(get_symbol_data(exchange_manager, symbol), min_timeframe)
+        market_begin = market_data[PriceIndexes.IND_PRICE_CLOSE.value][0]
+        market_end = market_data[PriceIndexes.IND_PRICE_CLOSE.value][-1]
+
+        if market_begin and market_end and market_begin > 0:
+            market_delta = market_end / market_begin - 1 if market_end >= market_begin \
+                else 1 - market_begin / market_end
+        else:
+            market_delta = 0
+
+        return market_delta
+
+    async def _init_evaluators(self):
```

Comment:
I think the following methods concerning the bot initialization with other module API should be in another dedicated class like `OctoBotBacktesting`.